src/components/core/geoChart/geoChartColor.ts

```diff
@@ -5,18 +5,21 @@ import isNumber = require("lodash/isNumber");
 import isFinite = require("lodash/isFinite");
 import { Execution } from "@gooddata/typings";
 import { DEFAULT_COLORS, getColorPalette } from "../../visualizations/utils/color";
-import { DEFAULT_PUSHPIN_COLOR_SCALE, DEFAULT_PUSHPIN_COLOR_VALUE } from "../../../constants/geoChart";
+import {
+    DEFAULT_PUSHPIN_COLOR_SCALE,
+    DEFAULT_PUSHPIN_COLOR_VALUE,
+    EMPTY_SEGMENT_ITEM,
+} from "../../../constants/geoChart";
 import { getHeaderItemName } from "../../../helpers/executionResultHelper";
 import { IObjectMapping, IPushpinColor } from "../../../interfaces/GeoChart";
 import { IColorLegendItem } from "../../visualizations/typings/legend";
 import isEmpty = require("lodash/isEmpty");
 
 const DEFAULT_SEGMENT_ITEM = "default_segment_item";
-const EMPTY_SEGMENT_ITEM = "empty_segment_item";
 const DEFAULT_COLOR_INDEX_IN_PALETTE = DEFAULT_PUSHPIN_COLOR_SCALE - 1;
 
 export function getColorIndexInPalette(value: number, min: number, max: number): number {
-    if (!isNumber(value) || value === min) {
+    if (!isNumber(value) || min === max || value === min) {
```

Comment:
Please make sure we have unit test for this new case

Reply:
added